include/pulsar/ClientConfiguration.h

```diff
@@ -320,6 +324,33 @@ class PULSAR_PUBLIC ClientConfiguration {
      */
     ClientConfiguration& setConnectionTimeout(int timeoutMs);
 
+    /**
+     * Set proxy-service url when client would like to connect to broker via proxy. Client must configure both
+     * proxyServiceUrl and appropriate proxyProtocol.
+     *
+     * Example: pulsar+ssl://ats-proxy.example.com:4443
+     *
+     * @param proxyServiceUrl proxy url to connect with broker
+     * @return
+     */
+    ClientConfiguration& setProxyServiceUrl(const std::string& proxyServiceUrl);
+
+    const std::string& getProxyServiceUrl() const;
+
+    /**
+     * Set appropriate proxy-protocol along with proxy-service url. Currently Pulsar supports SNI proxy
+     * routing.
+     *
+     * SNI routing:
+     * https://docs.trafficserver.apache.org/en/latest/admin-guide/layer-4-routing.en.html#sni-routing.
+     *
+     * @param proxyProtocol possible options (SNI)
+     * @return
+     */
+    ClientConfiguration& setProxyProtocol(const ProxyProtocol proxyProtocol);
```

Comment:
```suggestion
    ClientConfiguration& setProxyProtocol(ProxyProtocol proxyProtocol);
```

`const` for the parameter only works in the method implementation. When providing an API for users, it's better only to use `const` for references and pointers.

Reply:
addressed it.